src/anndata/experimental/backed/_xarray.py

```diff
@@ -128,10 +144,5 @@ def _remove_unused_categories_xr(
 
 
 @to_memory.register(Dataset2D)
-def to_memory(ds: Dataset2D, *, copy=False):
-    df = ds.to_dataframe()
-    index_key = ds.attrs.get("indexing_key", None)
-    if df.index.name != index_key and index_key is not None:
-        df = df.set_index(index_key)
-    df.index.name = None  # matches old AnnData object
-    return df
+def to_memory(ds: Dataset2D, *, copy=False) -> pd.DataFrame:
+    return ds.to_memory(copy=copy)
```

Comment:
could just be

```suggestion
to_memory.register(Dataset2D, Dataset2D.to_memory)
```

I think.